src/block_cache/block_chunkcache.c

```diff
@@ -20,6 +20,88 @@
 #define WT_CHUNK_OFFSET(chunkcache, offset) \
     (wt_off_t)(((size_t)offset / (chunkcache)->chunk_size) * (chunkcache)->chunk_size)
 
+/*
+ * __chunkcache_create_metadata_file --
+ *     Create the table that will persistently track what chunk cache content is on disk.
+ */
+static int
+__chunkcache_create_metadata_file(
+  WT_SESSION_IMPL *session, uint64_t capacity, unsigned int hashtable_size, size_t chunk_size)
+{
+    char cfg[128];
+    WT_RET(
+      __wt_snprintf(cfg, sizeof(cfg),
+        "app_metadata=\"version=1,capacity=%" PRIu64 ",buckets=%u,chunk_size=%" WT_SIZET_FMT "\","
+        "key_format=" WT_CC_KEY_FORMAT ",value_format=" WT_CC_VALUE_FORMAT,
+        capacity, hashtable_size, chunk_size));
+
+    return (__wt_session_create(session, WT_CC_URI, cfg));
+}
+
+/*
+ * __chunkcache_get_metadata_config --
+ *     If present, retrieve the on-disk configuration for the chunk cache metadata file. The caller
+ *     must only use *config if *found is true. The caller is responsible for freeing the memory
+ *     allocated into *config.
+ */
+static int
+__chunkcache_get_metadata_config(WT_SESSION_IMPL *session, bool *found, char **config)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    size_t len;
+    char *tmp;
+
+    *found = false;
+    *config = NULL;
+
+    WT_RET(__wt_metadata_cursor(session, &cursor));
+    cursor->set_key(cursor, WT_CC_URI);
+    WT_ERR_NOTFOUND_OK(cursor->search(cursor), true);
+
+    if (ret == 0) {
+        *found = true;
+
+        WT_ERR(cursor->get_value(cursor, &tmp));
+
+        len = strlen(tmp);
+        WT_ERR(__wt_calloc(session, 1, len + 1, config));
+        strcpy(*config, tmp);
+    } else
+        ret = 0;
+
+err:
+    WT_TRET(__wt_metadata_cursor_release(session, &cursor));
+    return (ret);
+}
+
+/*
+ * __chunkcache_verify_metadata_config --
+ *     Check that the existing chunk cache configuration is compatible with our current
+ *     configuration (and ergo, whether we can reuse the chunk cache contents).
+ */
+static int
+__chunkcache_verify_metadata_config(WT_SESSION_IMPL *session, char *md_config, uint64_t capacity,
+  unsigned int hashtable_size, size_t chunk_size)
+{
+    WT_DECL_RET;
+    char tmp[128];
+
+    WT_RET(__wt_snprintf(tmp, sizeof(tmp),
+      "app_metadata=\"version=1,capacity=%" PRIu64 ",buckets=%u,chunk_size=%" WT_SIZET_FMT "\"",
+      capacity, hashtable_size, chunk_size));
```

Comment:
Maybe this format should be defined in a constant, instead of appearing twice?

Reply:
Good idea, done in f332172.